core/src/main/java/com/linecorp/armeria/common/util/ShutdownHooks.java

```diff
@@ -42,6 +44,9 @@ public final class ShutdownHooks {
     private static final Map<AutoCloseable, Queue<Runnable>> autoCloseableOnShutdownTasks =
             new LinkedHashMap<>();
 
+    @GuardedBy("autoCloseableOnShutdownTasks")
```

Comment:
I believe that we need to add this `@GuardedBy` to the `autoCloseableOnShutdownTasks` field not to `reentrantLock` field.

Reply:
i fixed it as reviewed

now I am struggling with the error below. (Many errors were displayed😭, when  run `./gradlew --parallel build`)
```
/AbstractListenable.java:53: error: [GuardedBy] This access should be guarded by 'this.reentrantLock', which is not currently held
            updateListeners = this.updateListeners.toArray((Consumer<? super T>[]) EMPTY_LISTENERS);
                                  ^
```

[8ccb411](https://github.com/line/armeria/pull/4551/commits/8ccb4115b084fbc86277f032537289d7480f2e87)

Reply:
Let's inline methods. It seems like the errorprone doesn't analyze correctly when the lock is not directly used.

Reply:
I was worried, so I read [this](https://github.com/line/armeria/pull/4549#issuecomment-1336969325) and applied it in advance, But it didn't go well

I inlined it like below but same error still displayed
```java
    protected final void notifyListeners(T latestValue) {
        final Consumer<? super T>[] updateListeners;
        // lock() -> reentrantLock.lock();
        reentrantLock.lock();
        try {
            //noinspection unchecked
            updateListeners = this.updateListeners.toArray((Consumer<? super T>[]) EMPTY_LISTENERS);
        } finally {
            // unlock() -> reentrantLock.unlock();
            reentrantLock.unlock();
        }

        for (Consumer<? super T> listener : updateListeners) {
            listener.accept(latestValue);
        }
    }
```

Reply:
> I inlined it like below but same error still displayed

I've run it in my local and it worked. 😄 
Could you try again, please?

Reply:
I received the commit you sent and it works fine. thanks minwoox🙇🏻‍♂️🙇🏻‍♂️